spec/models/get_an_identity/user_spec.rb

```diff
@@ -0,0 +1,112 @@
+require "rails_helper"
+
+RSpec.describe GetAnIdentity::User do
+  describe "#find" do
+    # specs for find method
+    # stubs requests using webmock
+    # checks that the correct request is made
+    # checks that the response is properly loaded into a GetAnIdentity::User object
+    # checks that the correct error is raised when the response is not successful
+
+    let(:access_token) { SecureRandom.uuid }
+    let(:stubbed_url) { "https://example.com" }
+
+    let(:uid) { "123456" }
+
+    before do
+      allow(ENV).to receive(:fetch).with("TRA_OIDC_DOMAIN").and_return(stubbed_url)
+
+      stubbed_access_token = instance_double(GetAnIdentity::AccessToken, to_s: access_token)
+      allow(GetAnIdentity::AccessToken).to receive(:new).and_return(stubbed_access_token)
+
+      stub_request(:get, "#{stubbed_url}/api/v1/users/#{uid}")
+        .with(
+          headers: {
+            "Authorization" => "Bearer #{access_token}",
+          },
+        )
+        .to_return(
+          status: response_status_code,
+          body: response_body.to_json,
+          headers: { "Content-Type" => "application/json" },
+        )
+    end
+
+    context "when the response is successful" do
+      let(:response_uid) { uid }
+      let(:response_body) do
+        {
+          "userId" => response_uid,
+          "email" => "mail@example.com",
+          "firstName" => "Jane",
+          "lastName" => "Doe",
+          "dateOfBirth" => "1990-01-01",
+          "trn" => "1234567",
+          "mobileNumber" => "",
+          "mergedUserIds" => [],
+        }
+      end
+
+      let(:response_status_code) { 200 }
+
+      let(:loaded_user) do
+        GetAnIdentity::User.find("123456")
+      end
+
+      it "returns a user that responds to the correct attributes" do
+        expect(loaded_user.id).to eq uid
+        expect(loaded_user.uid).to eq response_uid
+        expect(loaded_user.email).to eq "mail@example.com"
+        expect(loaded_user.full_name).to eq "Jane Doe"
+        expect(loaded_user.date_of_birth).to eq Date.new(1990, 1, 1)
+        expect(loaded_user.trn).to eq "1234567"
+        expect(loaded_user.mobile_number).to eq ""
+        expect(loaded_user.merged_user_ids).to eq []
+        expect(loaded_user.raw).to eq response_body
+      end
+
+      context "when uid from response does not match the requested uid" do
```

Comment:
Does test actually test anything as couldnt work it out from the description and your stubbing the response so the response_uid is always going to be whatever you set it as right?

Reply:
It's stubbed yes, but it's making sure that if they do differ they get loaded appropriately, rather than one or the other being used for both. It'll allow you from the GetAnIdentity::User object you get back to see the ID you requested and the one you got back, which you could compare. Perhaps that's not the best approach though. It feels a little iffy to not have them be the same.

What do you think? Overwrite the ID with whatever you get back and leave the one requested to appear in the merged_user_ids array?

Reply:
Ok i think ive worked out my gripe with this... 😂 In the code your asking for a user with a particular ID which is used as the params so i was confused why the response userID would be any different? 

In the test your not really testing this as an edge case as such because the response will just come back with whatever your setting it as? ie its not like the functionality would change etc i guess.

Reply:
It is testing things, just because the API response is stubbed doesn't mean it isn't testing the code. If I go in and break my actual code, this test would fail.

For example if I change the code inside GetAnIdentity::User from `@uid = response["userId"]` to `@uid = @id` the spec goes red. 

It's an explicit test that we are exposing data from the API and not the data the user initially passed in. I agree that it's essentially a meaningless test since we know it works, but it makes it explicit in the specs that we know that these two can differ and that is expected. But I guess in the end, it's a relatively meaningless spec.

Reply:
I've gotten rid of the extra spec :) Probably right that it's not really testing much